op-service/sources/receipts.go

```diff
@@ -5,17 +5,277 @@ import (
 	"fmt"
 	"io"
 	"sync"
+	"time"
 
+	"github.com/ethereum-optimism/optimism/op-service/client"
 	"github.com/ethereum-optimism/optimism/op-service/sources/batching"
+	"github.com/ethereum-optimism/optimism/op-service/sources/caching"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/common/hexutil"
 	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
 	"github.com/ethereum/go-ethereum/rpc"
 	"github.com/ethereum/go-ethereum/trie"
 
 	"github.com/ethereum-optimism/optimism/op-service/eth"
 )
 
+type ReceiptsProvider interface {
```

Comment:
This is a bit of file jenga, but I wonder if it makes sense to put caching / basic / auto-selector layers in different files.

Reply:
I'm all for splitting stuff more into separate files! I just worked in the original file and was moving around stuff, but happy to separate.

Reply:
done